Code-Sleep-Python/tic-tac-toe/TicTacToe.py

```diff
@@ -0,0 +1,317 @@
+def magic():
+    n = 3
+    a = [[0 for i in range(0, n)] for j in range(0, n)]
+    k = 1
+    a[n // 2][n - 1] = k
+    k += 1
+    i = n // 2
+    j = n - 1
+    while k <= (n * n):
+        i -= 1
+        j += 1
+        if i == -1 and j != n:
+            i = n - 1
+        if j == n and i != -1:
+            j = 0
+        if i == -1 and j == n:
+            i = 0
+            j = n - 2
+        if a[i][j] != 0:
+            j -= 2
+            i += 1
+            if i == -1 and j == n:
+                i = 0
+                j = n - 2
+        a[i][j] = k
+        k += 1
+    return a
+a=magic()
+b=[]
+comp=[]
+human=[]
+for i in range(0,9):
+    b.append(0)
+def pos_win(arr):
+    flagh=0 # flag for human win
+    for ii in range(0, len(arr)):
+        for jj in range(ii + 1, len(arr)):
+            for kk in range(jj+1,len(arr)):
+                if ((arr[ii] + arr[jj] + arr[kk])==15):
+                    flagh=2
+                    break
+    return flagh
+def move_1():
+    flag=0
+    comp.append(5)
+    b[4]=1
+    return flag
+def move_2():
+    flag = 0
+    if b[4]==0:
+        comp.append(5)
+        b[4]=1
+    else:
+        b[1]=1
+        comp.append(2)
+    return flag
+def move_3():
+    flag = 0
+    if b[7]==0:
+        comp.append(8)
+        b[7]=1
+    else:
+        b[5]=1
+        comp.append(6)
+    return flag
+def move_4():
+    flag = 0
+    n=15-(human[0]+human[1])
+    if n>0 and n<=9 and b[n-1]==0:
+        b[n-1]=1
+        comp.append(n)
+    else:
+        done=0
+        for i in range(0,len(b)):
+            if (i+1)%2!=0 and b[i]==0:
+                b[i]=1
+                done=1
+                comp.append(i+1)
+                break
+        if done==0:
+            for i in range(0,len(b)):
+                if b[i]==0:
+                    b[i]=1
+                    comp.append(i+1)
+                    break
+    return flag
+def move_5():
+    flag = 0
+    n = 15 - (comp[0] + comp[1])
+    m1 = 15 - (human[0] + human[1])
+    if n>0 and n<=9 and b[n-1]==0:
+        b[n-1]=1
+        comp.append(n)
+        flag=1

```

Comment:
Please explain your code through comments and make it as self-explanatory as possible.

Also you need to show why your implementation is better than the one already provided here